### PR TITLE
Fixes #7152 - Gnesis now eats up your blood and then kills you

### LINKED ISSUE #7152: Coagulated Gnesis doesn't convert blood in bodies
### Labels

[BUG]

### Description

Coagulated Gnesis does not convert blood in humans/monkeys/etc

### Steps to reproduce

1. Ingest or inject coagulated gnesis.
2. Gnesis quickly decays into nothing if there are no non-blood reagents in the body, such as water.

### Expected Behavior

I expected the gnesis to convert blood into more gnesis, as is described on the wiki.

### Additional Information & Screenshots

On the wiki, gnesis is specifically described as being able to convert blood into more gnesis. I had to constantly fill my body with water just so I could even stay above the gib threshold. Probably a bug with chems not being able to interact with blood in the body. Might affect other chems that interact with blood, though I can't think of any right now.

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
Gnesis was processing as if the human body wasn't full of blood. Now it doesn't do that.

## Why's this needed? 
Gnesis now does what it says on the wiki, building up in your system until you explode into a flock bit.

### CODE REVIEW

**Comment by ThatFiveGuys** [comment]
Dont think this was a bug, having it eat up your blood seems super op and unfun to deal with. Gnesis scapels/pills would become a really good and unfun way to deal with antags, and one that would be really confusing and hard to deal with for new players

**Comment by amylizzle** [comment]
> Dont think this was a bug, having it eat up your blood seems super op and unfun to deal with. Gnesis scapels/pills would become a really good and unfun way to deal with antags, and one that would be really confusing and hard to deal with for new players

I mean, it isn't subtle and you have lots of time to deal with it even at a high starting dosage. 
I'm not particularly fussed either way though, but if this is preferred behaviour, someone should update the wiki to reflect that.

**Comment by ThatFiveGuys** [comment]
This assumes you 1. Know what gnesis the chem is and how it works (i seriously doubt a lot of players do) and 2. Know how to remove a hundred or so units of a chem from you (again assuming you know theres a chem in you), which yeah seems like itll just be used for validhunting new players who get a death that just leaves em confused and angry.

**Comment by amylizzle** [comment]
> This assumes you 1. Know what gnesis the chem is and how it works (i seriously doubt a lot of players do) and 2. Know how to remove a hundred or so units of a chem from you (again assuming you know theres a chem in you), which yeah seems like itll just be used for validhunting new players who get a death that just leaves em confused and angry.

I feel like you could say that about any poison though. Amantin in particular is markedly worse the gnesis in terms of subtlety and lethality. 

It takes about 5 minutes to get up to a lethal dose, with lots and lots of messages in teal italics about there being something wrong before blood loss effects start to kick in and you get *more* messages, before you finally develop actually debilitating symptoms. A trip to medbay and a semi-competent doctor should do the job.

**Comment by ThatFiveGuys** [comment]
The difference there is all those poisons cant be applied by hitting someone literally once. Also just go see a doctor doesnt apply when youre an antag that cant really disguise like wizard.

**Comment by Sbmhawk** [comment]
And wouldn't the gensis just eat the charcoal/meds that are used to treat it.

**Comment by amylizzle** [comment]
So we discussed this a little on discord, and I'm going to put a critical threshold in with this - below 100 units it decays as before because too dilute or something, above 100 units it builds up. 
Also I'm going to test that calomel can get rid of it, because that seems important.

**Comment by amylizzle** [comment]
Okay, updated:
Less than 50 units, the gnesis just decays - your kidney's filter it out or something, yay.
Above 50, it starts replacing your blood as it should.
However, the process is now slower and only affects *one chem at a time*. If you have five chems in you, it will pick one at random each tick and convert it, rather than all of them every tick. This means medicines now work, and are more effective if you have multiple in there (ie, charcoal and calomel).

Calomel is now strong enough to remove gnesis, though very advanced cases will require basically a calomel drip. It takes about 1u of calomel to deal with 2u of gnesis.

Note that having gnesis and another chem in your blood stream will convert the other chem as well, regardless of dosage, so if you have 50+ units of another chem and get stabbed with a gnesis scalpel, you're still gonna have a bad time.

File: code/modules/chemistry/Reagents-Misc.dm [hunk 1]
```diff
@@ -2056,16 +2056,23 @@ datum
...
-						if(reagent_id != id)
-							holder.remove_reagent(reagent_id, conversion_rate)
+					var/list/otherReagents = holder.reagent_list.Copy()
+					otherReagents -= src.id
+					if(ishuman(holder?.my_atom))
+						var/mob/living/carbon/human/H = holder.my_atom
+						if(H.blood_volume >= conversion_rate)
+							otherReagents += "blood_placeholder"
+
+					if(otherReagents.len > 0)
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-Misc.dm:2066]: length!

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/chemistry/Reagents-Misc.dm (hunk 1 above)
Comment (amylizzle) [reply code/modules/chemistry/Reagents-Misc.dm:2066]: dammit, I knew one of them was bad but I forgot which!

**Comment by Cthucky** [comment]
If I'm not mistaken, gnesis eating your blood was removed precisely because how absurdly easy it was to gib people with it.

**Comment by amylizzle** [comment]
> If I'm not mistaken, gnesis eating your blood was removed precisely because how absurdly easy it was to gib people with it.

Yeah, I'm finding myself wondering if it should actually gib you. Like, a flockbit bursting from your chest wouldn't necessarily have to gib you, just absurd brute + organ damage and kill you.

**Review by Tarmunora — APPROVED** [review]

**Comment by ZeWaka** [comment]
fine4now

**Pull request merged**